<?xml version="1.0"?>
<launch>

    <!-- Name of the VI sequence -->
    <arg name="result_save_path" default="/home/kailai/Documents/results/ctgaupro/"/>
    <arg name="vi_data_path" default="$(find ctgaupro)/dataVICalib/"/>

    <!-- Launch the camera imu calibration node -->
    <node pkg="ctgaupro" type="ctgaupro_gpvicalib" name="ctgaupro_gpvicalib" output="screen">
        
        <param name="auto_exit"  value="1"/>
        <param name="if_save_traj" value="0"/>
        <param name="traj_save_path" value="$(arg result_save_path)"/>
        <param name="data_path" value="$(arg vi_data_path)"/>

        <!-- Parameters for the Gaussian Process -->
        <param name="gpDt"       value="0.05"/>
        <param name="gpQr"       value="1.00"/>
        <param name="gpQc"       value="1.00"/>
        
        <!-- Parameters for the solver  -->
        <param name="SLIDE_SIZE"  value="2"/> <!-- How many knots to slide -->
        <param name="WINDOW_SIZE" value="20"/> <!-- How many knot length does the sliding window -->
        <param name="w_corner" value="1.0"/> <!-- Coefficients for TDOA residuals -->
        <param name="GYR_N" value="2000.0"/> <!-- Coefficients for IMU residuals -->
        <param name="GYR_W" value="100.0"/> <!-- Coefficients for IMU residuals -->
        <param name="ACC_N" value="100.0"/> <!-- Coefficients for IMU residuals -->
        <param name="ACC_W" value="100.0"/> <!-- Coefficients for IMU residuals -->
        <param name="corner_loss_thres" value="-1.0"/> <!-- Loss function for IMU residuals -->
        <param name="mp_loss_thres" value="100.0"/> <!-- Loss function for motion prior residuals -->

    </node>

   <!-- Visualize -->
    <node pkg="rviz" type="rviz" name="rviz_ui" output="log" required="true"
        args="-d $(find ctgaupro)/launch/ctgaupro_vicalib.rviz" />      

</launch>